src/web/components/Discussion.tsx

```diff
@@ -0,0 +1,138 @@
+};
+
+export const Discussion = ({
+    discussionApiUrl,
+    shortUrlId,
+    isCommentable,
+    pillar,
+    user,
+    discussionD2Uid,
+    discussionApiClientHeader,
+    enableDiscussionSwitch,
+    isAdFreeUser,
+    shouldHideAds,
+}: Props) => {
+    const [commentCount, setCommentCount] = useState<number>(0);
+    const [isClosedForComments, setIsClosedForComments] = useState<boolean>(
+        true,
+    );
+    const [commentPage, setCommentPage] = useState<number>();
+    const [commentPageSize, setCommentPageSize] = useState<25 | 50 | 100>();
+    const [commentOrderBy, setCommentOrderBy] = useState<
+        'newest' | 'oldest' | 'recommendations'
+    >();
+    const [isExpanded, setIsExpanded] = useState<boolean>(false);
+    const [hashCommentId, setHashCommentId] = useState<number | undefined>(
+        commentIdFromUrl(),
+    );
+    const hasCommentsHash = window?.location?.hash === '#comments';
+
+    const handlePermalink = (commentId: number) => {
+        window.location.hash = `#comment-${commentId}`;
+        const comment = window.document.getElementById(`comment-${commentId}`);
+        if (comment) {
+            // The comment was already on the page so just scroll to it.
+            comment.scrollIntoView();
+        }
+        setHashCommentId(commentId);
+        return false;
+    };
+
+    useEffect(() => {
+        const callFetch = async () => {
+            const response = await getDiscussion(discussionApiUrl, shortUrlId);
+            setCommentCount(
+                (response && response.discussion.commentCount) || 0,
+            );
+            setIsClosedForComments(
+                response && response.discussion.isClosedForComments,
+            );
+        };
+
+        if (isCommentable) {
+            callFetch();
+        }
+    }, [discussionApiUrl, shortUrlId, isCommentable]);
+
+    // Check the url to see if there is a comment hash, e.g. ...crisis#comment-139113120
+    // If so, make a call to get the context of this comment so we know what page it is
+    // on.
+    useEffect(() => {
+        if (hashCommentId) {
+            getCommentContext(discussionApiUrl, hashCommentId).then(
+                (context) => {
+                    setCommentPage(context.page);
+                    setCommentPageSize(context.pageSize);
+                    setCommentOrderBy(context.orderBy);
+                    setIsExpanded(true);
+                },
+            );
+        }
+    }, [discussionApiUrl, hashCommentId]);
+
+    useEffect(() => {
+        if (hasCommentsHash) {
+            setIsExpanded(true);
+        }
+    }, [hasCommentsHash]);
+
+    return (
+        <>
+            <Portal root="comment-count-root">
+                <CommentCount
+                    isCommentable={isCommentable}
+                    commentCount={commentCount}
+                    pillar={pillar}
+                    setIsExpanded={setIsExpanded}
+                />
+            </Portal>
```

Comment:
This actually feels quite nice to me from a 'keep relevant components together' POV. Nicer than a big bucket in App? Maybe we think about how we do Portals in general...